tests/unit/test_finetune_resources.py

```diff
@@ -335,3 +374,87 @@ def test_train_on_inputs_not_supported_for_dpo():
             training_method="dpo",
             train_on_inputs=True,
         )
+
+
+@patch("together.abstract.api_requestor.APIRequestor.request")
+def test_price_estimation_request(mocker):
+    test_data = [
+        {
+            "training_type": "lora",
+            "training_method": "sft",
+        },
+        {
+            "training_type": "lora",
+            "training_method": "dpo",
+        },
+        {
+            "training_type": "full",
+            "training_method": "sft",
+        },
+    ]
+    mocker.return_value = (
+        TogetherResponse(
+            data={
+                "estimated_total_price": 100,
+                "allowed_to_proceed": True,
+                "estimated_train_token_count": 1000,
+                "estimated_eval_token_count": 100,
+                "user_limit": 1000,
+            },
+            headers={},
+        ),
+        None,
+        None,
+    )
```

Comment:
Hm, it feels weird that we can't reuse `mock_request` here and have to define the mocking logic once again. I wonder if we can generalize somehow…